graylog2-web-interface/src/routing/AppGlobalNotifications.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { PluginStore } from 'graylog-web-plugin/plugin';
+
+import styles from './AppGlobalNotifications.css';
+
+const AppGlobalNotifications = React.createClass({
+  render() {
+    const globalNotifications = PluginStore.exports('globalNotifications')
+      .map((notification) => {
+        if (!notification.component) {
+          // eslint-disable-next-line no-console
+          console.error('Missing "component" for globalNotification plugin:', notification);
+          return null;
+        }
+        const Component = notification.component;
+        if (!notification.key) {
+          // eslint-disable-next-line no-console
+          console.warn('Missing "key" for globalNotification plugin:', notification);
+        }
+        return <Component key={notification.key} />;
+      })
+      .filter(component => !!component);
+
+    return (
+      <div className={`container-fluid ${styles.globalNotifications}`} id="global-notifications">
```

Comment:
Could we wrap the component's children in a react-bootstrap's `Row` and `Col`, avoiding to add styles in here? In the end the styling is just mimicking what those components do.

Reply:
I tried that before. The problem is that we get padding and margins even when there is no active notification.

Maybe you have an idea how to fix that when using Row/Col?

## With Row/Col
![image](https://user-images.githubusercontent.com/461/32845268-9bc9ed7a-ca24-11e7-9022-d57f28fb1d8a.png)

## With div and custom style

![image](https://user-images.githubusercontent.com/461/32845304-b1c5971e-ca24-11e7-85eb-457211692922.png)

Reply:
Oh, I see, that's why you are adding `:not(:empty)` to the CSS selector.

There is afaik no way for a parent to know if its children are rendering something or not. This is mostly due to how React works: you try to put all the logic you can upper in the tree and pass things down, not the other way around, which is what this situation would need.

Then I would suggest: why don't we leave the styling to the children? In the end we are not adding much styling in here, so we can also let plugin authors decide how the notification should look like. In the end they could also override anything we set in here.

Reply:
@edmundoa Good point. I will remove the styling and leave it to the children. :+1: